pkg/operator/key.go

```diff
@@ -0,0 +1,136 @@
+package operator
+
+import (
+	"crypto/ecdsa"
+	"crypto/elliptic"
+	"crypto/rand"
+	"fmt"
+	"math/big"
+)
+
+// Curve represents an elliptic curve name that denotes the curve used by
+// the operator keys.
+type Curve int
+
+// Enum that lists all the elliptic curves that can be used with the operator
+// keys.
+const (
+	Undefined Curve = iota
+	Secp256k1
+)
+
+// ParseCurve takes a curve name as string and parses it to a specific
+// operator.Curve enum instance.
+func ParseCurve(value string) (Curve, error) {
+	switch value {
+	case "undefined":
+		return Undefined, nil
+	case "secp256k1":
+		return Secp256k1, nil
+	}
+
+	return -1, fmt.Errorf("unknown curve: [%v]", value)
+}
+
+// String returns the string representation of a given operator.Curve
+// enum instance.
+func (c Curve) String() string {
+	switch c {
+	case Undefined:
+		return "undefined"
+	case Secp256k1:
+		return "secp256k1"
+	default:
+		panic("unknown curve")
+	}
+}
+
+// Holds the bit size of the supported elliptic curves.
+var curveBitSizes = map[Curve]int{
+	Secp256k1: 256,
+}
```

Comment:
Can we move `curveBitSizes` next to the const defining the type of the curve and denote all types of the curves used must have bit sizes specified? This way, we would be able to eliminate the error from the API of `MarshalUncompressed` and `MarshalCompressed` and simplify all its usages.

Reply:
Alternatively - and I like this option better - remove `curveBitSizes` altogether and assume all types available in `Curve` enum must be supported by Go `crypto/elliptic`. Then, we can use `crypto/elliptic` code in `MarshalUncompressed` and `MarshalCompressed` without rolling out our own code. I think it makes sense especially that `GenerateKeyPair` accepts `elliptic.Curve` as a parameter.

Reply:
See https://github.com/keep-network/keep-core/pull/3025/commits/c446d9ad0cb679a8eff5e92f46fa763e8b5e4a76